File: python/13_is_pangram.py
"""¿Es un pangrama? — Fácil · Strings

Devuelve True si la frase contiene todas las letras del alfabeto inglés,
ignorando mayúsculas y minúsculas. Los números, espacios y signos no cuentan
como letras.

Fuente: https://coding-challenges.dev/problems/python-es-pangrama
"""


def is_pangram(sentence: str) -> bool:
    alphabet = set("abcdefghijklmnopqrstuvwxyz")
    return alphabet.issubset(sentence.lower())

File: typescript/13-is-pangram.ts
// ¿Es un pangrama? — Fácil · Strings
// Devuelve true si contiene todas las letras del alfabeto inglés, sin importar
// mayúsculas y minúsculas. Números, espacios y signos no cuentan como letras.
// Fuente: https://coding-challenges.dev/problems/typescript-es-pangrama

export function isPangram(sentence: string): boolean {
  const letters = new Set(sentence.toLowerCase().match(/[a-z]/g));
  return letters.size === 26;
}
